sw/host/opentitanlib/src/transport/ti50emulator/mod.rs

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use log::{error, info};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use std::process;
+use std::rc::Rc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use crate::io::emu::Emulator;
+use crate::io::gpio::GpioPin;
+use crate::io::i2c::Bus;
+use crate::io::spi::Target;
+use crate::io::uart::Uart;
+use crate::transport::{
+    Capabilities, Capability, Result, Transport, TransportError, TransportInterfaceType,
+};
+
+mod emu;
+mod gpio;
+mod i2c;
+mod spi;
+mod uart;
+
+use crate::transport::ti50emulator::emu::Ti50SubProcess;
+
+pub struct Ti50Emulator {
+    inner: Rc<RefCell<Inner>>,
+}
+
+impl Ti50Emulator {
+    /// Create new instance of [`Ti50Emulator`] based on provided parameters.
+    pub fn open(
+        executable_directory: PathBuf,
+        executable: &String,
+        instance_prefix: &String,
+    ) -> anyhow::Result<Self> {
+        let tstamp = SystemTime::now().duration_since(UNIX_EPOCH)?;
+        let instance_name = format!(
+            "{}_{}_{}_{}",
+            instance_prefix,
+            process::id(),
+            tstamp.as_secs(),
+            tstamp.as_nanos()
+        );
+
+        let mut instance_directory = PathBuf::from("/tmp");
+        instance_directory.push(&instance_name);
+        let resource_directory = instance_directory.join("resources");
+        let runtime_directory = instance_directory.join("runtime");
+
+        info!("Initializing Ti50Emulator instance:{}", instance_name);
+        fs::create_dir(&instance_directory)?;
+        fs::create_dir(&resource_directory)?;
+        fs::create_dir(&runtime_directory)?;
+
+        Ok(Self {
+            inner: Rc::new(RefCell::new(Inner {
+                instance_directory: instance_directory.clone(),
+                _resource_directory: resource_directory,
+                _runtime_directory: runtime_directory,
+                _executable_directory: executable_directory,
+                _executable: Some(executable.clone()),
+                emulator: None,
+                spi_map: HashMap::new(),
+                gpio_map: HashMap::new(),
+                i2c_map: HashMap::new(),
+                uart_map: HashMap::new(),
+            })),
+        })
+    }
+}
+
+impl Drop for Ti50Emulator {
+    fn drop(&mut self) {
+        info!(
+            "Clenup Ti50Emulator instance directory: {}",
+            self.inner.borrow().instance_directory.display()
+        );
+        if let Err(e) = fs::remove_dir_all(&self.inner.borrow().instance_directory) {
+            error!("Can't remove instance directory error: {}", e)
+        }
+    }
+}
+
+/// Structure representing internal state of emulator
+struct Inner {
+    /// Path of parent directory representing instance.
+    instance_directory: PathBuf,
+    /// Path to a directory holding unmodified files (for example flash image).
+    _resource_directory: PathBuf,
```

Comment:
Here too, If not used, why are they here?

Reply:
Those fields should be added only once you begin using them. I'd take a look at https://google.github.io/eng-practices/review/developer/small-cls.html to understand why this practice is important.

Reply:
Ok. I will delete unused variables.

It will be good to add something simillar (or link to this document) to OpenTitan documentations.
We have some overall guidelines heare: 
[OT contributing](https://docs.opentitan.org/doc/project/detailed_contribution_guide/#working-with-pull-requests)